firecracker/microvm.py

```diff
@@ -400,6 +433,17 @@ async def teardown(self):
         if self.stderr_task:
             self.stderr_task.cancel()
 
+        if self.mounted_fs:
+            logger.debug("Waiting for one second for the VM to shutdown")
```

Comment:
Is there no signal you could use from Firecracker, like the PID of the VM or something like that? It would be better to wait until the VM is down and not an arbitrary delay.

Reply:
Moved to a new issue so we can merge this.